packages/flutter_tools/lib/src/android/android_workflow.dart

```diff
@@ -389,12 +389,15 @@ class AndroidLicenseValidator extends DoctorValidator {
         ),
       );
 
+      final Stream<List<int>> stderrBroadcast = process.stderr.asBroadcastStream();
```

Comment:
using a broadcast stream worries me (because debugging stream issues is not easy). what about instead if you make a custom listener for `process.stderr` that both forwards data to `_stdio.stderr` and adds it to `stderrLines`?

Reply:
I cannot submit a code suggestion for a block this big, but this passes locally for me:

```dart
      try {
        final StreamSubscription<List<int>> stderrSub = process.stderr.listen((List<int> bytes) {
          _stdio.stderr.add(bytes);
          stderrLines.add(utf8.decoder.convert(bytes));
        });
        await Future.wait<void>(<Future<void>>[
          _stdio.addStdoutStream(process.stdout),
          stderrSub.asFuture<void>(),
        ]);
        await stderrSub.cancel();
      } on Exception catch (err, stack) {

```

Reply:
I commited my own take on this at the same time you sent this. Feel free to take a look. I know very little about Dart `Stream`s, so I don't know what's idiomatic here.

Reply:
your's is cleaner, LGTM